javalin/src/main/java/io/javalin/websocket/JavalinWsServlet.kt

```diff
@@ -59,6 +60,19 @@ class JavalinWsServlet(val config: JavalinConfig, private val httpServlet: Javal
             config.inner.accessManager.manage({ ctx -> ctx.req.setAttribute(upgradeAllowedKey, true) }, upgradeContext, entry.permittedRoles)
             if (req.getAttribute(upgradeAllowedKey) != true) throw UnauthorizedResponse() // if set to true, the access manager ran the handler (== valid)
             req.setAttribute(upgradeContextKey, upgradeContext)
+
+            // hhh, 2020.5.13 we should select one protocol according to IETF RFC 6455
+            val s = req.getHeader(WebSocketConstants.SEC_WEBSOCKET_PROTOCOL)
+            if (s != null) {
+                val protocolNames = s.split(',').map{it.trim()}.filter { it!="" }
+                if (protocolNames.isNotEmpty()) {
+                    // which protocol?  Maybe should given it in configuration
+                    val preferProtocol = "mqtt"
+                    val selectedProtocol = protocolNames.firstOrNull { it==preferProtocol }
+                    res.setHeader(WebSocketConstants.SEC_WEBSOCKET_PROTOCOL, selectedProtocol ?: protocolNames[0])
```

Comment:
We should not set the header unless the server actually supports the protocol requested. This should be provided at during configuration of Javalin, and should be up to the application to know how to handle that protocol.

Reply:
Yes, maybe needing:

  config.setWebSocketSubProtocol("mqtt")
or:
  config.setWebSocketSubProtocols(listOf("mqtt", "gRpc"))

Reply:
Ok, I'll improve it by adding config functions.